test/test_functional_tensor.py

```diff
@@ -868,6 +798,84 @@ def test_perspective_interpolation_warning(tester):
         tester.assertTrue(res1.equal(res2))
 
 
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('dt', [None, torch.float32, torch.float64, torch.float16])
+@pytest.mark.parametrize('size', [32, 26, [32, ], [32, 32], (32, 32), [26, 35]])
+@pytest.mark.parametrize('max_size', [None, 34, 40, 1000])
+@pytest.mark.parametrize('interpolation', [BILINEAR, BICUBIC, NEAREST])
+def test_resize(device, dt, size, max_size, interpolation, tester):
+
+    torch.manual_seed(12)
+    script_fn = torch.jit.script(F.resize)
+    tensor, pil_img = tester._create_data(26, 36, device=device)
+    batch_tensors = tester._create_data_batch(16, 18, num_samples=4, device=device)
+
+    if dt == torch.float16 and device == "cpu":
+        # skip float16 on CPU case
+        return
+
+    if dt is not None:
+        # This is a trivial cast to float of uint8 data to test all cases
+        tensor = tensor.to(dt)
+        batch_tensors = batch_tensors.to(dt)
+
+    if max_size is not None and isinstance(size, Sequence) and len(size) != 1:
+        return  # unsupported, see assertRaises below
+
+    resized_tensor = F.resize(tensor, size=size, interpolation=interpolation, max_size=max_size)
+    resized_pil_img = F.resize(pil_img, size=size, interpolation=interpolation, max_size=max_size)
+
+    assert resized_tensor.size()[1:] == resized_pil_img.size[::-1]
+
+    if interpolation not in [NEAREST, ]:
+        # We can not check values if mode = NEAREST, as results are different
+        # E.g. resized_tensor  = [[a, a, b, c, d, d, e, ...]]
+        # E.g. resized_pil_img = [[a, b, c, c, d, e, f, ...]]
+        resized_tensor_f = resized_tensor
+        # we need to cast to uint8 to compare with PIL image
+        if resized_tensor_f.dtype == torch.uint8:
+            resized_tensor_f = resized_tensor_f.to(torch.float)
+
+        # Pay attention to high tolerance for MAE
+        tester.approxEqualTensorToPIL(
+            resized_tensor_f, resized_pil_img, tol=8.0,
+            msg="{}, {}, {}".format(size, max_size, interpolation)
+        )
+
+    if isinstance(size, int):
+        script_size = [size, ]
+    else:
+        script_size = size
+
+    resize_result = script_fn(
+        tensor, size=script_size, interpolation=interpolation, max_size=max_size
+    )
+    assert_equal(resized_tensor, resize_result, msg="{}, {}".format(size, interpolation))
+
+    tester._test_fn_on_batch(
+        batch_tensors, F.resize, size=script_size, interpolation=interpolation, max_size=max_size
+    )
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+def test_resize_asserts(device, tester):
+
+    tensor, pil_img = tester._create_data(26, 36, device=device)
+
+    # assert changed type warning
+    with tester.assertWarnsRegex(UserWarning, r"Argument interpolation should be of type InterpolationMode"):
```

Comment:
let's go full pytest and use

```py
with pytest.raises(UserWarning, match=...)
```

Also I think that only the first line should be in the context manager, so that it's clear which line produces the warning, although it's NBD.